pupil-spa/src/app/login/login.component.ts

```diff
@@ -22,10 +23,15 @@ export class LoginComponent implements OnInit {
     private router: Router,
     private questionService: QuestionService,
     private warmupQuestionService: WarmupQuestionService,
-    private registerInputService: RegisterInputService
+    private registerInputService: RegisterInputService,
+    private checkRecoveryService: CheckRecoveryService
   ) { }
 
   ngOnInit() {
+    const hasUnfinishedCheck = this.checkRecoveryService.hasUnfinishedCheck();
+    if (hasUnfinishedCheck) {
+      this.router.navigate(['check']);
```

Comment:
is this navigating back to the recovery page?

Reply:
@GuyHarwood yes, it will navigate back to the submission pending screen and perform the end of check process.